pkg/util/deployment/deployment.go

```diff
@@ -139,7 +139,14 @@ func GetAvailablePodsForRCs(c client.Interface, rcs []*api.ReplicationController
 	readyPodCount := 0
 	for _, pod := range allPods {
 		if api.IsPodReady(&pod) {
-			readyPodCount++
+			// Check if we've passed minReadySeconds since LastTransitionTime
+			// If so, this pod is ready
+			for _, condition := range pod.Status.Conditions {
+				if minReadySeconds <= 0 || condition.LastTransitionTime.Add(time.Duration(minReadySeconds)*time.Second).Before(time.Now()) {
```

Comment:
Oh, right, we don't need to wait. If the pod is not available yet, deployment controller will call GetAvailablePodsForRCs again in the next pull.